core/services/flux_monitor.go

```diff
@@ -290,8 +328,57 @@ func (p *PollingDeviationChecker) CurrentPrice() decimal.Decimal {
 	return p.currentPrice
 }
 
+// CurrentRound returns the latest round.
+func (p *PollingDeviationChecker) CurrentRound() *big.Int {
+	return new(big.Int).Set(p.currentRound)
+}
+
+// RespondToNewRound takes the round broadcasted in the log event, and responds
+// on-chain with an updated price.
+// Only invoked by the CSP consumer on the single goroutine for thread safety.
+func (p *PollingDeviationChecker) RespondToNewRound(log eth.Log) error {
+	requestedRound, err := models.ParseNewRoundLog(log)
+	if err != nil {
+		return err
+	}
+
+	// skip if requested is not greater than current.
+	if requestedRound.Cmp(p.currentRound) < 1 {
+		logger.Infow(fmt.Sprintf(
+			"deviation checker ignoring new round request: requested %s <= current %s",
+			requestedRound.String(),
+			p.currentRound.String()),
+			"address", log.Address.Hex(),
+			"jobID", p.initr.JobSpecID,
+		)
+		return nil
+	}
+
+	logger.Infow(fmt.Sprintf(
+		"deviation checker responding to new round request: requested %s > current %s",
+		requestedRound.String(),
+		p.currentRound.String()),
+		"address", log.Address.Hex(),
+		"jobID", p.initr.JobSpecID,
+	)
+	p.currentRound = requestedRound
```

Comment:
Safe to assume updating `p.currentRound` is thread safe because of CSP?

Reply:
Yes, with the annoying exception of `func (p *PollingDeviationChecker) CurrentRound() *big.Int` which is only used in tests. I will fix it to be an `ExportedCurrentRound()` helper. Thanks for pointing this out.

Reply:
I attempted to make a `CurrentRound`, `CurrentPrice` getter CSP friendly here: https://github.com/smartcontractkit/chainlink/pull/2071/commits/6e3de155b6705c5d25c132edf97b263e7653a93b

```golang
func (p *PollingDeviationChecker) CurrentRound() *big.Int {
	rchan := make(chan *big.Int)
	p.currentRoundRequest <- rchan
	return <-rchan
}
```

but ran into a snag. We need the CSP consumer to be going, and that's invoked in `Start`. But our test checks price,round before `Start` but after `Initialize`, so it wasn't suitable for our needs. Went with the thread unsafe `ExportedCurrentRound`.

Feel free to discuss with me further in slack.